spark_log_parser/loaders/__init__.py

```diff
@@ -319,7 +327,9 @@ def extract(self, filepath: Path, file_stream: BufferedIOBase = None) -> Iterato
                     # We may see files like {name}.zip.gz/.json.gz, so make sure we handle that appropriately
                     if len(suffixes) == 1:
                         wrapped = FileChunkStreamWrapper(file)
-                        yield filepath, self.read_file_stream(wrapped)
+                        fs_iter = self.read_file_stream(wrapped)
+                        yield filepath, fs_iter
+                        exhaust_iterator(fs_iter)
```

Comment:
Is this only necessary for zip archives? In which case could we get away with only exhausting chunks in `read_zip_archive`?

Reply:
It would be required for `.tar` archives as well, I believe, but yeah, I had this same thought. I think the tricky part is that when we are extracting a file from an archive, it may go through a few more `extract` calls before we get the actual `Iterator` of the file, e.g. for files in the archive that may be gzipped (technically we support nested `.zip` archives, but that's more incidental than intentional, I believe, so perhaps not a large concern).

I'm sure we could figure something out for this, but another consideration for _non_-archive eventlogs I think would be network connections - if we are processing a streaming response, and just ... don't consume the rest of the stream, I believe that's just going to sit open until something finally kills it. I haven't tested this, so of course this is a little hypothetical at the moment, but this kind of feels analogous to opening a file descriptor and just never closing it, instead relying on GC to fix it for you. And since botocore, for example, is blocking, that could lead to some resource consumption issues down the line.

Reply:
> if we are processing a streaming response, and just ... don't consume the rest of the stream, I believe that's just going to sit open until something finally kills it.

Good point. However it should be the sole responsibility of whatever takes an external resources to also relinquish it.

Based on the tgz function not doing anything special when skipping a file seems like it doesn't have the same complication as the unzip stream. Could the complexities of the zip stream be pushed further down to the function that creates it? And if any other format requires some special management hopefully it can likewise contain it.

Reply:
I tested this by swapping the order we test archives in our unit tests, and indeed, `tarfile` seems to handle this more transparently! Which is great. Just about to push a commit that moves this into `read_zip_archive`, as it does indeed appear that we don't need it elsewhere